src/tiled/automapper.cpp

```diff
@@ -469,10 +469,22 @@ void AutoMapper::autoMap(QRegion *where)
                 TileLayer *dstTileLayer = dstLayer->asTileLayer();
                 if (dstTileLayer)
                     dstTileLayer->erase(region);
-                else
+                else {
+                    // Region has to be put into object space
+                    const int w = mMapDocument->map()->tileWidth();
+                    const int h = mMapDocument->map()->tileHeight();
+                    const QRect& tilesRect = region.boundingRect();
+                    QRect objRect;
+                    objRect.setTop(tilesRect.top() * h);
+                    objRect.setLeft(tilesRect.left() * w);
+                    objRect.setBottom((tilesRect.bottom() + 1) * h);
+                    objRect.setRight((tilesRect.right() + 1) * w);
+                    QRegion objRegion;
+                    objRegion += objRect;
```

Comment:
Hmm, this seems not exactly correct in two ways:

* It should use the `MapRenderer` to convert the region into object space, though I'm not sure if the current API suffices for this. Maybe it could convert each corner of each rectangle in the region and include the bounding rect of the result into the new region.

* It's converting a bounding rect of the entire region rather than converting each rectangle separately, causing a potentially larger area to be erased than was intended?